syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.AFS";
option java_outer_classname = "AFS";
option objc_class_prefix = "HLW";

package FS;

service AFS {
  rpc GetAttr (GetAttrRequest) returns (GetAttrResponse) {}
  rpc Open (OpenRequest) returns (OpenResponse) {}
  rpc Close (CloseRequest) returns (CloseResponse) {}
  rpc Mkdir (MkdirRequest) returns (MkdirResponse) {}
}


message CloseRequest {
  string path = 1;
  bytes  fileData = 2;
}

message CloseResponse {
  int32 status = 1;
}


message OpenRequest {
  string path = 1;
  int32 flags = 2;
}

message OpenResponse {
  int32 status = 1;
  bytes fileData = 2; 
}

message GetAttrRequest {
  string path = 1;
}

message GetAttrResponse {
  int32 status = 1;
  bytes statBuf = 2;
}

message MkdirRequest {
  string path = 1;
  int32 mode = 2;
}

message MkdirResponse {
  int32 status = 1;
}
